Feat(eos_cli_config_gen)!: Add new data model for ip_name_server with modified vrfs key (#6199)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

| `generate_default_config` | - |
| `generate_device_documentation` | `eos_cli_config_gen_documentation.enable` |
| `ip_name_server_groups[].ip_domain_list` | `ip_name_server_groups[].ip_domain_lists` |

> | `ip_name_servers` | `ip_name_server` |

Reviewer: Apart from old and new key lets add and example about how to use new model

Author: added, we can add other daat-model entries under this heading.
@gmuloc is it okay to keep like this ?
also, should we remove this one now `### `ip_name_servers[].vrf` is required to generate name servers configuration` line no. 857 (since we have changed the model)

Reviewer 2: yes please rmeove the outdated porting guide entry. good catch thanks!

Author: Removed!

+    vrf: default
```

### Data-models changed to handle VRF entries

| Old data-model | New data-model |
| -------------- | -------------- |
| ip_name_servers | ip_name_server |

#### ip_name_servers data-model is replaced with the new data-model ip_name_server

In AVD 5.x, the data model `ip_name_servers` needed multiple entries for same VRF with multiple domain ip addresses.

> Starting with AVD 6.0.0,  `ip_name_servers` has been replaced by `ip_name_server` data-model, where the VRF is now the primary key with `servers.ip_address` list to handle multiple ip addresses.

Reviewer 3: ```suggestion
Starting with AVD 6.0.0, `ip_name_servers` has been replaced with `ip_name_server` data-model, where the VRF name is now the primary key with `servers.ip_address` list to handle multiple ip addresses.
```

+    vrf: default
```

### Data-models changed to handle VRF entries

| Old data-model | New data-model |
| -------------- | -------------- |
| ip_name_servers | ip_name_server |

> #### ip_name_servers data-model is replaced with the new data-model ip_name_server

Reviewer: ```suggestion
#### `ip_name_servers` data model is replaced by the new `ip_name_server` data model
```

In AVD 5.x, the data model `ip_name_servers` needed multiple entries for same VRF with multiple domain ip addresses.

> Starting with AVD 6.0.0, `ip_name_servers` has been replaced with `ip_name_server` data-model, where the VRF name is now the primary key with `servers.ip_address` list to handle multiple ip addresses.

Reviewer: ```suggestion
#### `ip_name_servers` data model is replaced by the new `ip_name_server` data model

In AVD 5.x, the `ip_name_servers` data model rendered multiple entries for the same VRF when configuring multiple name-server IP addresses.

Starting with AVD 6.0.0, `ip_name_servers` has been replaced by the `ip_name_server` data model. In the new structure, the VRF name is the primary key, and multiple name-server addresses are defined under servers.ip_address, allowing all IPs for a given VRF to be grouped together.
```
